airflow/providers/amazon/aws/sensors/redshift.py

```diff
@@ -37,18 +37,25 @@ def __init__(
         self,
         *,
         cluster_identifier: str,
-        target_status: str = 'available',
+        target_status: RedshiftClusterStates = RedshiftClusterStates.AVAILABLE,
         aws_conn_id: str = 'aws_default',
         **kwargs,
     ):
         super().__init__(**kwargs)
         self.cluster_identifier = cluster_identifier
-        self.target_status = target_status
+        self.target_status = (
+            target_status
+            if isinstance(target_status, RedshiftClusterStates)
+            else RedshiftClusterStates(str(target_status))
+        )
+
```

Comment:
why not simply

```suggestion
        self.target_status = RedshiftClusterStates(target_status)
```

it does  work even if target_status  is the enum

Reply:
also, if you intend to be permissive you should hint above with

```python
target_status: Union[RedshiftClusterStates, str] = RedshiftClusterStates.AVAILABLE,
```

Reply:
i didn't know it would work even if `target_status` is the enum :) thanks for that